server.js

```diff
@@ -2,31 +2,81 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 
+
 // If you're using one of our datasets, uncomment the appropriate import below
 // to get started!
 // 
 // import goldenGlobesData from './data/golden-globes.json'
 // import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
+import booksData from './data/books.json'
+//import netflixData from './data/netflix-titles.json'
 // import topMusicData from './data/top-music.json'
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 8088
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+//Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Endpoints:  /books, /books/:id, /ratings, authors/:author')
 })
 
+//List all the books in the api with /books and query for title
+
+app.get('/books', (req, res) => {
+  const { title } = req.query
+  const bookTitle = booksData.filter((item) =>
+    item.title.toString().toLowerCase().includes(title))
+
+  if (bookTitle.length > 0) {
+    res.json(bookTitle)
+
+  } else if (title && bookTitle.length === 0) {
+    res.status(404).json({ message: 'Book title not found' })
+
+  } else {
+    res.json(booksData)
+  }
+})
+
+
+
+//Show one book using an :id as a placeholder for the bookID number in the data
+app.get('/books/:id', (req, res) => {
+  const { id } = req.params
+  const bookId = booksData.find((item) => item.bookID === +id)
+  if (bookId) {
+    res.json(bookId)
+  } else {
+    res.status(404).json({ message: 'Not found' })
+  }
+})
+
+
+//Sort books after rating, highest to lowest 
+app.get('/ratings', (req, res) => {
+  const bestRated = booksData.sort((a, b) => b.average_rating - a.average_rating);
+  res.json(bestRated)
+})
+
+
+//Get all books by one author (example: https://express-book-api-hannaruter.herokuapp.com/authors/rowling)
+app.get('/authors/:author', (req, res) => {
+  const author = req.params.author;
```

Comment:
When pointing to a specific resource in the endpoint, such as an author, it's really good to use a path parameter like you have here.

Your `author` path parameter filters a result instead of "identifying" a result.

For instance 
`GET /authors/dan` returns books from 
Dan Brown and Daniel Abraham.  

A **path parameter** is a good option if you are using it to point to a specific author, however **query parameters** are more optimal when filtering results which could return more than one different author.